messaging/src/main/java/org/axonframework/configuration/LifecycleSupportingConfiguration.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.configuration;
+
+/**
+ * An interface providing both {@link NewConfiguration} and {@link LifecycleOperations} support.
+ * <p>
+ * Ensures correct encapsulation when configuration instances get constructed.
+ *
+ * @author Allard Buijze
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public interface LifecycleSupportingConfiguration extends NewConfiguration, LifecycleOperations {
```

Comment:
I am not sure about this intermediary interface. What's the upside compared to letting the ones that support both, implement both? :thinking:

Reply:
I intend to get rid of this interface entirely as part of a follow-up PR that'll adjust the life cycle registration logic.